### PR TITLE
kill existing archive on force cancel

### PR DESCRIPTION
**Pull request by lrivera-oicr**
Dispose if there is a current build running.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -80,6 +80,7 @@ spec:
             when {
                 anyOf {
                     branch 'develop'
+                    branch 'kill_archive_build'
```
Comment (caravinci) [inline Jenkinsfile:83]:
```suggestion
```

File: Jenkinsfile [hunk 2]
```diff
@@ -125,6 +126,7 @@ spec:
...
```
Comment (caravinci) [inline Jenkinsfile:129]:
```suggestion
```

File: src/main/java/org/cancogenvirusseq/singularity/components/hoc/ArchiveBuildRequestToArchive.java [hunk 3]
```diff
@@ -58,15 +58,21 @@ public Flux<Archive> apply(ArchiveBuildRequest archiveBuildRequest) {
         .flatMap(
             uploadObjectId ->
                 withArchiveBuildRequestContext(
-                    archiveBuildRequestCtx -> {
-                        archiveBuildRequestCtx.getArchive().setObjectId(uploadObjectId);
-                        archiveBuildRequestCtx.getArchive().setStatus(ArchiveStatus.COMPLETE);
+                    archiveBuildRequestCtx ->
+                        archivesRepo
+                            .findByArchiveObject(archiveBuildRequestCtx.getArchive())
+                            .filter(archive -> ArchiveStatus.BUILDING.equals(archive.getStatus()))
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/components/hoc/ArchiveBuildRequestToArchive.java:64]: Mind explaining the intent for these two lines in the scope of this PR?

**Review by caravinci — APPROVED** [review]
Looks great! Just left a comment to ask for clarification, and added the commits to remove the automatic branch deployment, when the PR has been tested and ready to squash+merge.

**Review by caravinci — APPROVED** [review]
LGTM 👍 

Squash it away!

File: src/main/java/org/cancogenvirusseq/singularity/components/hoc/ArchiveBuildRequestToArchive.java (hunk 3 above)
Comment (leoraba) [reply src/main/java/org/cancogenvirusseq/singularity/components/hoc/ArchiveBuildRequestToArchive.java:71]: to make sure that the archive object current status is BUILDING before change it to COMPLETE.

**Pull request merged**